app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java

```diff
@@ -72,10 +68,151 @@ public static SessionStore get() {
+        public static final String SECURE_CONNECTION_ERROR_URL = "resource://android/assets/html/error/secure-connection-failed.html";
+
+        public static final String NET_ERROR_TYPE = "about:neterror";
+        public static final String CERT_ERROR_TYPE = "about:certerror";
+
+        public enum GeckoErrorType {
+            UNKNOWN_ERROR,
+            NET_ERROR,
+            CERT_ERROR
+        }
+
+        public static final String DNSNOTFOUND_ERROR_SUBTYPE = "dnsNotFound";
+        public static final String NSSFAILURE2_ERROR_SUBTYPE = "nssFailure2";
+        public static final String FILENOTFOUND_ERROR_SUBTYPE = "fileNotFound";
+        public static final String NSSBADCERT_ERROR_SUBTYPE = "nssBadCert";
+
+        public enum GeckoErrorSubType {
+            UNKNOWN,
+            DNS_NOT_FOUND,
+            NSS_FAILURE2,
+            FILE_NOT_FOUND,
+            NSS_BAD_CERT
+        }
+
+        private GeckoErrorType mErrorType;
+        private GeckoErrorSubType mErrorSubType;
+        private String mErrorUri;
+        private String mErrorDescription;
+        private String mRedirectUri;
+        private String mOriginalUri;
+
+        public GeckoErrorType getErrortype() {
+            return mErrorType;
+        }
+
+        public GeckoErrorSubType getErrorSubType() {
+            return mErrorSubType;
+        }
+
+        public String getErrorUri() {
+            return mErrorUri;
+        }
+
+        public String getErrorDescription() {
+            return mErrorDescription;
+        }
+
+        public String getRedirectUri() {
+            return mRedirectUri;
+        }
+
+        public String getOriginalUri() {
+            return mOriginalUri;
+        }
+
+        public static GeckoError parse(String aUri) {
+            GeckoError geckoError = new GeckoError();
+            geckoError.mOriginalUri = aUri;
+
+            int parseStartPos;
+            if (aUri.startsWith(NET_ERROR_TYPE)) {
+                 parseStartPos = NET_ERROR_TYPE.length() + 1;
+                geckoError.mErrorType = GeckoErrorType.NET_ERROR;
+
+            } else if (aUri.startsWith(CERT_ERROR_TYPE)) {
+                parseStartPos = CERT_ERROR_TYPE.length() + 1;
+                geckoError.mErrorType = GeckoErrorType.CERT_ERROR;
+
+            } else {
+                geckoError.mErrorType = GeckoErrorType.UNKNOWN_ERROR;
+                return geckoError;
+            }
+
+            try {
+                String query = aUri.substring(parseStartPos);
+                String[] pairs = query.split("&");
+                for (String pair : pairs) {
+                    int idx = pair.indexOf("=");
+                    String key = URLDecoder.decode(pair.substring(0, idx), "UTF-8");
+                    String value = URLDecoder.decode(pair.substring(idx + 1), "UTF-8");
+                    if (key.equalsIgnoreCase("e")) {
+                        if (value.equalsIgnoreCase(DNSNOTFOUND_ERROR_SUBTYPE)) {
```

Comment:
was this code pulled from Firefox? if so, which file - just for my own curiosity?